nautobot/extras/jobs.py

```diff
@@ -477,14 +478,17 @@ def as_form(cls, data=None, files=None, initial=None, approval_view=False):
         try:
             job_model = JobModel.objects.get_for_class_path(cls.class_path)
             dryrun_default = job_model.dryrun_default if job_model.dryrun_default_override else cls.dryrun_default
-            task_queues = job_model.task_queues if job_model.task_queues_override else cls.task_queues
+            # Initialize job_queue choices
+            form.fields["_job_queue"] = DynamicModelChoiceField(
+                queryset=JobQueue.objects.filter(jobs=job_model),
+                query_params={"jobs": [job_model.pk]},
+                required=False,
+                help_text="The job queue to route this job to",
+                label="Job queue",
+            )
         except JobModel.DoesNotExist:
             logger.error("No Job instance found in the database corresponding to %s", cls.class_path)
             dryrun_default = cls.dryrun_default
-            task_queues = cls.task_queues
```

Comment:
Do we need to create a stub `_job_queue` form field in this exception case?

Reply:
What would be the choices in the form field in that case? Just default or None

Reply:
I don't remember what the code does here in this case - do we still present a form to the user even though it can't be run?

Reply:
Hmmm we do still present the same form but we are using the class attribute `cls.task_queues`.
But without the job_model, the queryset for `_job_queue` form field becomes unfilterable. What I am doing rn is something similar to that.
```
         try:
            job_model = JobModel.objects.get_for_class_path(cls.class_path)
            dryrun_default = job_model.dryrun_default if job_model.dryrun_default_override else cls.dryrun_default
            job_queue_queryset = JobQueue.objects.filter(jobs=job_model)
            job_queue_params = {"jobs": [job_model.pk]}
        except JobModel.DoesNotExist:
            logger.error("No Job instance found in the database corresponding to %s", cls.class_path)
            dryrun_default = cls.dryrun_default
            pk_list = []
            for queue in cls.task_queues:
                job_queue, _ = JobQueue.objects.get_or_create(
                            name=queue, defaults={"queue_type": JobQueueTypeChoices.TYPE_CELERY}
                        )
                pk_list.append(job_queue.pk)
            job_queue_queryset = JobQueue.objects.filter(pk__in=pk_list)
            job_queue_params = {"id": pk_list}


        # Initialize job_queue choices
        form.fields["_job_queue"] = DynamicModelChoiceField(
            queryset=job_queue_queryset,
            query_params=job_queue_params,
            required=False,
            help_text="The job queue to route this job to",
            label="Job queue",
        )
```

Reply:
I guess I'm trying to understand why we're presenting a form in this case instead of giving an error message and sending the user back to the jobs list. In this case since we can't find the `job_model`, it's not something we can actually run, right?

Reply:
Yeah so the original code is:
```
       except JobModel.DoesNotExist:
            logger.error("No Job instance found in the database corresponding to %s", cls.class_path)
            dryrun_default = cls.dryrun_default
            task_queues = cls.task_queues

        # Update task queue choices
        form.fields["_task_queue"].choices = task_queues_as_choices(task_queues)

        if cls.supports_dryrun and (not initial or "dryrun" not in initial):
            # Set initial "dryrun" checkbox state based on the Meta parameter
            form.fields["dryrun"].initial = dryrun_default
        if not settings.DEBUG:
            form.fields["_profile"].widget = forms.HiddenInput()

        # https://github.com/PyCQA/pylint/issues/3484
        if cls.field_order:  # pylint: disable=using-constant-test
            form.order_fields(cls.field_order)

        if approval_view:
            # Set `disabled=True` on all fields
            for _, field in form.fields.items():
                field.disabled = True

        return form
```
When we do not find the Job in the database, we get those attributes (`dryrun_default`, `task_queues`) using the Job class meta attributes and return the form regardless.

Reply:
Take out the try...except..block and just grab the job_model straight up

Reply:
Looks like this broke `test_job_variables` because the jobs that it defines and uses are local to the test and are not registered in the database. I was afraid that might happen. Sorry! Maybe as a fix could explicitly register those jobs as part of the test?